include/domain_bridge/topic_bridge_options.hpp

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021, Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DOMAIN_BRIDGE__TOPIC_BRIDGE_OPTIONS_HPP_
+#define DOMAIN_BRIDGE__TOPIC_BRIDGE_OPTIONS_HPP_
+
+#include <memory>
+
+#include "rclcpp/callback_group.hpp"
+
+#include "domain_bridge/visibility_control.hpp"
+
+namespace domain_bridge
+{
+
+/// Options for bridged topics.
+class TopicBridgeOptions
+{
+public:
+  /// Constructor.
+  /**
+   *  Default values:
+   *
+   *    - callback_group = nullptr (node's default)
+   */
+  DOMAIN_BRIDGE_PUBLIC
+  TopicBridgeOptions() = default;
+
+  /// Destructor.
+  DOMAIN_BRIDGE_PUBLIC
+  virtual
+  ~TopicBridgeOptions() = default;
+
+  /// Copy constructor.
+  DOMAIN_BRIDGE_PUBLIC
+  TopicBridgeOptions(const TopicBridgeOptions & other) = default;
+
+  /// Assignment operator.
+  DOMAIN_BRIDGE_PUBLIC
+  TopicBridgeOptions &
+  operator=(const TopicBridgeOptions & other) = default;
+
+  /// Get callback group associated with the topics pub/sub pair.
+  DOMAIN_BRIDGE_PUBLIC
+  std::shared_ptr<rclcpp::CallbackGroup>
+  callback_group() const;
```

Comment:
Having read this far, I'm not clear on the usage of this callback_group. I assume its part of the implementation and not for the user.

Reply:
This is meant as a configuration point for the user. This relates to the executor discussion above https://github.com/ros2/domain_bridge/pull/1#discussion_r586759175. It enables a more advance users to customize the execution model.

I work on improving the docs.

Reply:
That would be helpful. I'd like to understand how one could use this feature. For example, could one implement some kind of filtering proxy model using this? Could log messages that have been bridged, potentially block messages (although I'm not sure why one would do this). Looking for some ideas on how it would be used.